grpc/apiproxy/proxies/default.xml

```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<!--
+ Copyright 2023 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+
+<ProxyEndpoint name="default">
+  <Description/>
+  <FaultRules/>
+  <PreFlow name="PreFlow">
+    <Request>
+      <Step>
+        <Name>VA-Verify-API-key</Name>
+      </Step>
+      <Step>
+        <Name>AM-Remove-API-Key-Param</Name>
+      </Step>
+    </Request>
+    <Response/>
+  </PreFlow>
+  <PostFlow name="PostFlow">
+    <Request/>
+    <Response/>
+  </PostFlow>
+  <Flows/>
+  <HTTPProxyConnection>
+    <BasePath>/helloworld.Greeter/SayHello</BasePath>
```

Comment:
The basepath should be something like /v1/samples/grpc

Reply:
Maybe is my lack of knowledge around gRPC or grpcurl, but it breaks when you use a different basepath in the proxy than the service name:

``` 
grpcurl -import-path $HOME/cloudshell_open/apigee-samples/grpc/grpc-backend/grpc/examples/protos -proto helloworld.proto -d '{"name":"Guest"}' $ENV_GROUP_HOSTNAME_GRPC:443 v1/samples/grpc

Error invoking method "v1/samples/grpc": target server does not expose service "v1/samples"
```

Reply:
@ssvaidyanathan 
[Link](https://learn.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?view=aspnetcore-7.0#calling-grpc-services-hosted-in-a-sub-directory) for context: 
The path component of a gRPC channel's address is ignored when making gRPC calls. For example, GrpcChannel.ForAddress("https://localhost:5001/ignored_path") won't use ignored_path when routing gRPC calls for the service.

The address path is ignored because gRPC has a standardized, prescriptive address structure. A gRPC address combines the package, service and method names: https://localhost:5001/PackageName.ServiceName/MethodName.

Reply:
hmm - can we deploy the grpc service with /v1/samples :)

else I will have to create an exception in the linting rule which I dont want to :)

Reply:
I totally get that... I'll ping Geir to see if he has an idea, because based on all these GitHub discussions it seems like they have a strong opinion on keeping everything under root...and there doesn't seem to be an easy workaround. Or am I missing something? Here's another example: 

https://github.com/grpc/grpc-dotnet/issues/110